src/util/misc.ts

```diff
@@ -46,27 +46,26 @@ export function touch(paths: string[] | string) {
   });
 }
 
-export function getFullUrl(webUrlHost: string, url: string, baseUrl: string) {
-  if (typeof url !== 'string' || !url) {
-    throw new Error(`Expected url to be a string, got [${url}] instead`);
-  }
+export function getFullUrl(url: string, baseUrl: string) {
```

Comment:
`baseUrl` should be a URL object. Considering that is is almost always caleed with `mw.base` that should save us the parsing. But that mean you need to update `Mediawiki.base`. To me, `Mediawiki.base` should be called `Mediawiki.baseUrl` and should be an object. I don't really understand why we don't have moved in to URL object like the others?!